Clusterwide installation: ClusterRole needs create verb for statefulsets.apps

**What did you do to encounter the bug?**
Steps to reproduce the behavior:
Follow instructions, configure operator to watch all namespace, install operator and try to deploy a first example database in a different namespace.

1. set watch "*"
1. kubectl create namespace mongodb-operator
1. kubectl apply -f deploy/clusterwide
1. kubectl apply -k config/rbac -n mongodb-operator 
1. kubectl apply -f config/manager/manager.yaml -n mongodb
1. kubectl create namespace mongodb-tests
1. kubectl apply -k config/rbac -n mongodb-tests
1. create example database. See example code below.

**What did you expect?**
Database gets created in the mongodb-tests namespace.

**What happened instead?**
Nothing.

An error appeared in the mongodb operator pod:

```bash
2021-07-30T10:01:53.068Z        ERROR   controllers/mongodb_status_options.go:84        Error deploying MongoDB ReplicaSet: error creating/updating StatefulSet: error creating/updating StatefulSet: statefulsets.apps is forbidden: User "system:serviceaccount:mongodb-operator:mongodb-kubernetes-operator" cannot create resource "statefulsets" in API group "apps" in the namespace "mongodb-tests"
```

**Operator Information**
 - Operator Version: current master branch
 - commit 39ddc054ee4b0fff9fef35f8c6b04effd2de53cf (HEAD -> master, origin/master, origin/HEAD)
 - MongoDB Image used: no changes


**Kubernetes Cluster Information**
 - Distribution: Rancher RKE v2.5
 - kubernetes Version v1.20.8
 - Image Registry location (quay, or an internal registry) - not relevant

**Additional context**

Code for example mongodb deployment in the mongodb-tests namespace:

```yaml
---
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: example-mongodb
spec:
  members: 3
  type: ReplicaSet
  version: "4.2.6"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: my-user
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: my-user-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib

# the user credentials will be generated from this secret
# once the credentials are generated, this secret is no longer required
---
apiVersion: v1
kind: Secret
metadata:
  name: my-user-password
type: Opaque
stringData:
  password: securityIsImportant!
```


According to the error message (*cannot create resource "statefulsets" in API group "apps" in the namespace "mongodb-tests"*), a possible workaround would be to just add the create verb in the statefulsets.apps api group just like that:
 
```yaml
- apiGroups:
  - apps
  resources:
  - statefulsets
  verbs: 
  - create
  - ...
  -
```

--- a/deploy/clusterwide/role.yaml
+++ b/deploy/clusterwide/role.yaml
@@ -8,6 +8,7 @@ rules:
   resources:
   - services
   - configmaps
+  - secrets
   verbs:
   - create
   - delete
@@ -20,7 +21,8 @@ rules:
   - apps
   resources:
   - statefulsets
-  verbs: 
+  verbs:
+  - create
   - delete
   - get
   - list

--- a/deploy/clusterwide/role_binding.yaml
+++ b/deploy/clusterwide/role_binding.yaml
@@ -4,6 +4,7 @@ metadata:
   name: mongodb-kubernetes-operator
 subjects:
 - kind: ServiceAccount
+  namespace: <your-namespace>
   name: mongodb-kubernetes-operator
 roleRef:
   kind: ClusterRole

--- a/docs/install-upgrade.md
+++ b/docs/install-upgrade.md
@@ -62,19 +62,21 @@ To configure the Operator to watch resources in other namespaces:
                  value: "*"
    ```
 
-2. Run the following command to create cluster-wide roles and role-bindings in the default namespace:
+2. Modify the [clusterRoleBinding](../deploy/clusterwide/role_binding.yaml) namespace value for the serviceAccount `mongodb-kubernetes-operator` to the namespace in which the operator is deployed.
+
+3. Run the following command to create cluster-wide roles and role-bindings in the default namespace:
 
    ```sh
    kubectl apply -f deploy/clusterwide
    ```
-3. For each namespace that you want the Operator to watch, run the following
+4. For each namespace that you want the Operator to watch, run the following
    commands to deploy a Role, RoleBinding and ServiceAccount in that namespace:
 
    ```sh
    kubectl apply -k config/rbac --namespace <my-namespace>
    ```
 
-4. [Install the operator](#procedure).
+5. [Install the operator](#procedure).
 
 ### Configure the MongoDB Docker Image or Container Registry
 

